### DISCUSSION TITLE
Getting Device Connection Quality

### DISCUSSION THREAD

**Discussion by QaunainM** [camaraproject/DeviceStatus#134] [Ideas]
Hi all, are there any plans for Device Status to get the ability to show the quality (level) of data and call signal, either at a time point or over a time series (maybe a subscription would be best for the time series?)

This would enable developer to implement things like QoD when the Device Signal Quality stay below a certain threshold for more than x seconds.

I saw there is a Device Quality Indicator proposal by @NoelWirzius , but does this one refer to it becoming its own API? https://github.com/camaraproject/WorkingGroups/pull/380/files

**Comment by akoshunyadi** [comment]
Hi @QaunainM , the idea of the new API Device Quality Indicator was presented in the last DeviceStatus subproject call. If it will be realized, it might be done within the Device Status working group.

**Reply by QaunainM** [reply]
Sounds good, @NoelWirzius had a mini meeting with me and a few others about the idea for a few APIs that can help solve these use cases

**Comment by QaunainM** [comment]
@NoelWirzius @akoshunyadi does it look like a [Device Quality Insights API](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/94044185/DeviceQualityIndicator) will be created or will some of that vision be merged into Device Status (Reachability)?

**Reply by akoshunyadi** [reply]
@QaunainM DeviceQualityIndicator is an own repository within the Device Status subproject. However at this moment we are lacking on use cases and having a reduced interest on the operator's side.

**Reply by QaunainM** [reply]
Thanks for the update about DeviceQualityIndicator 

One separate topic I think we have use cases for [this topic](https://github.com/camaraproject/DeviceStatus/discussions/147#discussioncomment-10834480 ), especially with larger volume companies is this, specifically 'The ability to detect if a phone number is virtual (VoIP) or traditional (not VoIP)

https://github.com/camaraproject/DeviceStatus/discussions/147#discussioncomment-10834480
